RFC-0011-mixed-case-identifier-support.md

```diff
@@ -0,0 +1,220 @@
+# **RFC-0010 for Presto**
+
+## Mixed case identifiers
+
+Proposers
+
+* Reetika Agrawal
+
+## Summary
+
+Improve Presto's identifier (schema, table & column names) handling to align with SQL standards, ensuring better interoperability with case-sensitive
+and case-normalizing databases while minimizing SPI-breaking changes.
+
+## Background
+
+Presto treats all identifiers as case-insensitive, normalizing them to lowercase. This creates issues when
+querying databases that are case-sensitive (e.g., MySQL, PostgreSQL) or case-normalizing to uppercase (e.g., Oracle,
+DB2). Without a standard approach, identifiers might not match the actual names in the underlying data sources, leading
+to unexpected query failures or incorrect results. Additionally, inconsistent handling of delimited and non-delimited
+identifiers across different connectors further complicates cross-engine compatibility.
+
+The goal here is to improve interoperability with storage engines by aligning identifier handling with SQL standards
+while ensuring a seamless user experience. Ideally, the change should be implemented in a way that minimizes
+breaking changes to the SPI, i.e. allowing connectors to adopt the new approach without significant.
+
+### Goals
+
+- Align Presto’s identifier handling with SQL standards to improve interoperability with case-sensitive and
+  case-normalizing databases.
+- Ensure consistent handling of delimited ("identifier") and non-delimited (identifier) identifiers across different connectors.
+- Minimize SPI-breaking changes to maintain backward compatibility for existing connectors.
+- Introduce a mechanism for connectors to define their own identifier normalization behavior.
+- Allow identifiers to retain their original case where necessary, preventing unexpected query failures.
+- Ensure Access Control SPI can correctly normalize identifiers.
+- Preserve a seamless user experience while making these changes.
+
+### Proposed Plan
+
+Connectors handle identifiers in three ways:
+
+1. SQL-Compliant (e.g., Oracle, DB2)
+  - Delimited identifiers (e.g., "MyTable") keep their original case.
+  - Non-delimited identifiers (e.g., mytable) are converted to uppercase.
+
+2. Case-Sensitive (e.g., PostgreSQL)
+  - Delimited identifiers keep their original case.
+  - Non-delimited identifiers may be converted to lowercase or another case.
+
+3. Case-Insensitive (e.g., Hive)
+  - Delimited and non-delimited identifiers are treated the same.
+  - Identifiers may be automatically converted to a specific case.
+
+Presto's behavior includes:
+
+- Delimited identifiers ("Identifier") and non-delimited identifiers (identifier) are converted to lowercase by default
+  unless a connector enforces a specific behavior.
```

Comment:
> unless a connector enforces a specific behavior.

Where do we enforce this currently?

Reply:
Its going to be in ConnectorMetadata API implementation here - https://github.com/prestodb/rfcs/pull/36/files#diff-217aa5922002c9dabc8df724c3930f3fa969a98881495ad2648be620fdb50407R105